src/main/java/com/palantir/docker/compose/configuration/DaemonEnvironmentValidator.java

```diff
@@ -21,18 +21,26 @@
 import static com.palantir.docker.compose.configuration.EnvironmentVariables.DOCKER_TLS_VERIFY;
 import static java.util.stream.Collectors.joining;
 
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import com.google.common.collect.ImmutableSet;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public final class DaemonEnvironmentValidator {
+public final class DaemonEnvironmentValidator implements EnvironmentValidator {
 
     private static final Set<String> ILLEGAL_VARIABLES = ImmutableSet.of(DOCKER_TLS_VERIFY, DOCKER_HOST, DOCKER_CERT_PATH);
+    private static final Supplier<DaemonEnvironmentValidator> SUPPLIER = Suppliers.memoize(
+            () -> new DaemonEnvironmentValidator());
+
+    public static DaemonEnvironmentValidator instance() {
+        return SUPPLIER.get();
```

Comment:
See comment on `DockerType` for more info. Awkwardness of having a supplier for both validators is because we're basically providing behavior here and wiring it up in the enum (conceptually, we're defining classes in the enum and attaching behavior from these other interfaces using objects). The more idiomatic approach would probably be to use a class hierarchy, and then this implementation would move to a single class.

I would likely prefer the class hierarchy approach, but started with this approach since it more closely mirrors existing code and current codebase seems to bias towards breaking out behavior into separate objects/classes over consolidating in methods.